gbdk-lib/examples/cross-platform/text_basic/src/main.c

```diff
@@ -0,0 +1,76 @@
+#include <gbdk/platform.h>
+#include <stdint.h>
+#include "Font.h"
+
+
+uint8_t GetCharacterVRamTile(char character) {
+
+    uint8_t vramTile=0;
+
+
+    // Char's can be interpreted as integers
+    // We don't need to map every alpha-numeric character
+    // We can use basic math to simplify A-Z and 0-9
+    if(character>='a'&&character<='z')vramTile = (character-'a')+1;
+    else if(character>='A'&&character<='Z')vramTile = (character-'A')+1;
+    else if(character>='0'&&character<='9')vramTile = (character-'0')+27;
+    else if(character=='!')vramTile = 37;
+    else if(character==':')vramTile = 38;
+    else if(character=='?')vramTile = 39;
+    else if(character=='/')vramTile = 40;
+    else if(character=='=')vramTile = 41;
+    else if(character==',')vramTile = 42;
+    else if(character=='.')vramTile = 43;
+    else if(character=='<')vramTile = 44;
+    else if(character=='>')vramTile = 45;
```

Comment:
(This applies for all of the examples with the same code)

- It's a little smaller code wise and perhaps a bit more efficient to use a case/select for the misc tiles. Perhaps more so as font characters are added. At some point a recode table might become more efficient, but this is probably good to start with.
- Having spaces in the if statements and the code they execute, and aligning the formatting makes it a bit easier to read and understand them.
```C
    if      (character >= 'a' && character <= 'z') vramTile = (character - 'a') + 1;
    else if (character >= 'A' && character <= 'Z') vramTile = (character - 'A') + 1;
    else if (character >= '0' && character <= '9') vramTile = (character - '0') + 27;
    else {
        switch(character) {
            case '!': vramTile = 37; break;
            case ':': vramTile = 38; break;
            case '?': vramTile = 39; break;
            case '/': vramTile = 40; break;
            case '=': vramTile = 41; break;
            case ',': vramTile = 42; break;
            case '.': vramTile = 43; break;
            case '<': vramTile = 44; break;
            case '>': vramTile = 45; break;
        }
    }
```